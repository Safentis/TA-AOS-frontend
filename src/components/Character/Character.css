.character {
    box-shadow: 
        .1rem .1rem .1rem var(--main-dk-grey-color),
        -.1rem -.1rem .1rem var(--main-dk-grey-color);
    border-radius: .5rem;
    border-top-left-radius: 0;
    border-top-right-radius: 0;
    display: grid;
    grid-template-rows: 20rem 1fr;
}

.character__image {
    width: 100%;
    height: 100%;
}

.character__description {
    padding: .5rem 1rem;
}