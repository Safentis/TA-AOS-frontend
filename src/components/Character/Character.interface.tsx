export interface Props {
  className?: string;
  gender: string;
  species: string;
  status: string;
  image: string;
  name: string;
}
