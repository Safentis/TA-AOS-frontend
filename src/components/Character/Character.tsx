import React, { FC } from 'react';

import './Character.css';
import { Props } from './Character.interface';

const Character: FC<Props> = ({
  className = '',
  gender,
  species,
  status,
  image,
  name,
}): React.ReactElement => {
  return (
    <section className={'character ' + className}>
      <div className="character__avatar">
        <img className="character__image" src={image} />
      </div>
      <ul className="description-character character__description">
        <li className="description-character__item">
          <h2>{name}</h2>
        </li>
        <li className="description-character__item">{gender}</li>
        <li className="description-character__item">{species}</li>
        <li className="description-character__item">{status}</li>
      </ul>
    </section>
  );
};

export default Character;
