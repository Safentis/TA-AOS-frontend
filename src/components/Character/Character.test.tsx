import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import Character from './Character';

import { Props } from './Character.interface';

const setUp = (props: Props) => {
  return shallow(<Character {...props} />);
};

describe('<Character />', () => {
  let component: ShallowWrapper<
    any,
    Readonly<{}>,
    React.Component<{}, {}, any>
  >;
  let props: Props = {
    className: 'custom-style',
    species: 'Human',
    gender: 'male',
    status: 'alive',
    image: 'www.domain.com',
    name: 'morty',
  };

  beforeEach(() => {
    component = setUp(props);
  });

  it('component does match to the snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});