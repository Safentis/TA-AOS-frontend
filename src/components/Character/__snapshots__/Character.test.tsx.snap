// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Character /> component does match to the snapshot 1`] = `
<section
  className="character custom-style"
>
  <div
    className="character__avatar"
  >
    <img
      className="character__image"
      src="www.domain.com"
    />
  </div>
  <ul
    className="description-character character__description"
  >
    <li
      className="description-character__item"
    >
      <h2>
        morty
      </h2>
    </li>
    <li
      className="description-character__item"
    >
      male
    </li>
    <li
      className="description-character__item"
    >
      Human
    </li>
    <li
      className="description-character__item"
    >
      alive
    </li>
  </ul>
</section>
`;
